geopy/geocoders/googleplaces.py

```diff
@@ -0,0 +1,217 @@
+            For example, if the input is 'Google' and the offset is 3, the service will match on 'Goo'.
+            If no offset is supplied, the service will use the whole term.
+        :type offset: int
+
+        :type location: :class:`geopy.point.Point`, list or tuple of (latitude,
+            longitude), or string as "%(latitude)s, %(longitude)s"
+
+        :param int radius:  The distance (in meters) within which to return place results.
+            Setting a radius biases results to the indicated area, but may not fully restrict results to the specified area.
+
+        :param string language: The language in which to return results.
+
+        :param string types: The types of place results to return.
+            Possible values are "geocode", "address", "establishment", "(regions)" or "(cities)"
+            If no type is specified, all types will be returned.
+
+        :param dict components: A grouping of places to which you would like to restrict your results.
+            Currently, you can use components to filter by country.
+            For example: components=country:fr would restrict your results to places within France.
+        """
+        autocomplete_predictions = self.autocomplete(query, timeout, offset, location, radius, language, types, components)
+        return [self.place_details(place.get('place_id'), language) for place in autocomplete_predictions]
+
+    def parse_details(self, details_page):
+        status = details_page.get('status')
+        if status != "OK":
+            self._check_status(status)
+            return None
+        return details_page.get('result')
+
+    # for getting only details about a specific place
+    def place_details(self, placeid, language):
+        detail_params = {}
+        detail_params['key'] = self.api_key
+        if not placeid:
+            return []
+        detail_params['placeid'] = placeid
+        if language:
+            detail_params['language'] = language
+        details_url = "?".join((self.details_api, urlencode(detail_params)))
+        detail_result = self.parse_details(self._call_geocoder(details_url))
+        formatted_address = detail_result['formatted_address']
+        latitude = detail_result['geometry']['location']['lat']
+        longitude = detail_result['geometry']['location']['lng']
+        return Location(formatted_address, (latitude, longitude), detail_result)
+
+    # for getting only autocomplete endpoint results
+    def autocomplete(self, query,
+                     timeout=None,
+                     offset=None,
+                     location=None,
+                     radius=None,
+                     language=None,
+                     types=None,
+                     components=None):
+
+        autocomplete_params = {
+            'input': self.format_string % query,
+        }
+
+        if self.api_key:
+            autocomplete_params['key'] = self.api_key
+
+        if offset:
```

Comment:
`is not None` (if the called explicitly passes `0` or `""`, for whatever reason, you should pass that to google)